pkg/inframetadata/internal/hostmap/hostmap.go

```diff
@@ -0,0 +1,89 @@
+package hostmap
+
+import (
+	"fmt"
+	"sync"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.uber.org/multierr"
+
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/inframetadata/gohai"
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/inframetadata/payload"
+)
+
+// HostMap stores a map from hostnames to host metadata payloads.
+// Host metadata payloads can be updated with the information from a given resource.
+// At any time, the internal map can be extracted and the map will be cleared out.
+type HostMap struct {
+	// mutex for the host map.
+	mutex sync.Mutex
+	// hosts map
+	hosts map[string]payload.HostMetadata
+}
+
+// New creates a new HostMap
+func New() (*HostMap, error) {
+	return &HostMap{
+		mutex: sync.Mutex{},
+		hosts: make(map[string]payload.HostMetadata),
+	}, nil
+}
+
+// Update the information about a given host by providing a resource.
+// The function reports:
+//   - Whether the information about the `host` has changed
+//   - Any non-fatal errors that may have occurred during the update
+//
+// Partial modifications will still be applied even with non-fatal errors.
+func (m *HostMap) Update(host string, res pcommon.Resource) (changed bool, err error) {
+	hm := payload.HostMetadata{
```

Comment:
Many times metadata is called `md` (e.g. in gRPC package).

Reply:
Renamed